app/src/main/java/com/keylesspalace/tusky/StatusListActivity.kt

```diff
@@ -262,6 +276,24 @@ class StatusListActivity : BottomSheetActivity(), HasAndroidInjector {
                     }
                 }
             )
+
+            if (filterCreateSuccess) {
+                updateTagMuteState(true)
+                Snackbar.make(binding.root, getString(R.string.muting_hashtag_success_format, tag), Snackbar.LENGTH_LONG).apply {
+                    setAction(R.string.action_view_filter) {
+                        val intent = if (mutedFilter != null) {
+                            Intent(this@StatusListActivity, EditFilterActivity::class.java).apply {
+                                putExtra(EditFilterActivity.FILTER_TO_EDIT, mutedFilter)
+                            }
+                        } else {
+                            Intent(this@StatusListActivity, FiltersActivity::class.java)
+                        }
```

Comment:
```suggestion
                        val intent = mutedFilter?.let {
                            Intent(this@StatusListActivity, EditFilterActivity::class.java).apply {
                                putExtra(EditFilterActivity.FILTER_TO_EDIT, it)
                            }
                        } ?: Intent(this@StatusListActivity, FiltersActivity::class.java)
```

Better would be to add static factory methods to `FiltersActivity` and `EditFilterActivity` that return the intent (like, e.g., `AccountListActivity.newIntent()`)

Reply:
Will do if we keep this.

See "normally open filter overview or the specific one" discussion above.